### PR TITLE
(BIDS-1334) Concurrent map iteration and map write - bigtable_eth1.go…

### PR DESCRIPTION
**Pull request by recy21**
…#L213

### CODE REVIEW

File: db/bigtable_eth1.go [hunk 1]
```diff
@@ -2123,31 +2123,39 @@ func (bigtable *Bigtable) GetInternalTransfersForTransaction(transaction []byte,
...
-	for address := range names {
-		address := address
-		g.Go(func() error {
-			name, err := bigtable.GetAddressName([]byte(address))
-			if err != nil {
-				return err
+		g := new(errgroup.Group)
+		g.SetLimit(25)
+		for address := range forNames {
+			address := address
```
Comment (D13ce) [inline db/bigtable_eth1.go:2136]: I know this was not changed by you, but the naming of this variable is VERY confusing (as the name is already used).
Feel free to change or to ignore.

File: db/bigtable_eth1.go [hunk 2]
```diff
@@ -2123,31 +2123,39 @@ func (bigtable *Bigtable) GetInternalTransfersForTransaction(transaction []byte,
...
+					mux.Lock()
+					names[address] = name
+					mux.Unlock()
+					return nil
+				})
+			} else {
+				mux.Lock()
+				names[address] = ""
+				mux.Unlock()
 			}
```
Comment (D13ce) [inline db/bigtable_eth1.go:2152]: I don't understand this else-block:
Why would you put something at `names[""]`?
Please share your thoughts.

**Review by D13ce — COMMENTED** [review]
One open question (see https://github.com/gobitfly/eth2-beaconchain-explorer/pull/1729/files?diff=unified&w=0#r1064421795), once this is tackled everything's fine.

File: db/bigtable_eth1.go (hunk 1 above)
Comment (recy21) [reply db/bigtable_eth1.go:2136]: Yes, I agree, but this seems to be the Golang way of doing it. :man_shrugging:

File: db/bigtable_eth1.go (hunk 2 above)
Comment (recy21) [reply db/bigtable_eth1.go:2152]: - To keep current / previous behavior
- There will be a map read later in the code for index "", even if names["random-index"] will be "" anyway, I feels more accurate to me.

But sure, the else block can be removed without any issue, I guess

**Review by D13ce — COMMENTED** [review]
lgtm

**Pull request merged**